app/src/test/kotlin/com/intive/picover/parties/repository/PartiesRepositoryTest.kt

```diff
@@ -77,8 +85,49 @@ class PartiesRepositoryTest : ShouldSpec(
 			every { databaseReference.snapshots } returns flowOf(dataSnapshot)
 
 			shouldThrowExactly<NoSuchElementException> {
-				tested.partyById(1).first()
+				tested.partyById("1").first()
 			}
 		}
+
+		should("add party with correct id") {
+			val newKey = "newKey"
+			val partyRemote = PartyRemote(
+				id = "id",
+				title = "title",
+				description = "desc",
+			)
+			val modifiedPartyRemote = PartyRemote(
+				id = newKey,
+				title = "title",
+				description = "desc",
+			)
+			val childDatabaseReference: DatabaseReference = mockk {
+				every { key } returns newKey
+				coEvery { setValue(modifiedPartyRemote).await() } returns mockk()
+				// every { setValue(modifiedPartyRemote) } returns mockk {
+				// 	every { isComplete } returns true
+				// 	every { exception } returns null
+				// 	every { isCanceled } returns false
+				// 	every { result } returns mockk()
+				// 	coEvery { await() } returns mockk()
+				// }
+			}
+			every { databaseReference.push() } returns childDatabaseReference
+
+			tested.addParty(partyRemote)
+
+			coVerify {
+				childDatabaseReference.setValue(modifiedPartyRemote).await()
+			}
+		}
+
+		should("throw an exception when setValue fails") {
+			val partyRemote = PartyRemote(id = "id", title = "title", description = "desc")
+			coEvery { databaseReference.setValue(any<PartyRemote>()) } throws Exception("error")
```

Comment:
here we can probably inline mocking for push and set value :)

Reply:
Test has been changed.